### PR TITLE
[Doc Update] document hostRedirectUrl for external OAuth flow

### PR DESCRIPTION
**Pull request by onmomo**
As of In `TeamsJS` version 2.11.0 or later, apps must provide a third url parameter, `hostRedirectUrl`. ([Source](https://github.com/MicrosoftDocs/msteams-docs/blob/033009d16afee721d8fca679e139d3b38badcb2b/msteams-platform/tabs/how-to/using-teams-client-library.md?plain=1#L86)) This PR should help developers to implement the external auth flow, correctly.

Unfortunately, there are several clients that don't seem to implement this but this is not part of the docs
* https://github.com/OfficeDev/microsoft-teams-library-js/issues/1696#issuecomment-1979154860

### Caused by
* https://github.com/OfficeDev/microsoft-teams-library-js/issues/1696#issuecomment-1971904831

### Relates to docs PR
* https://github.com/MicrosoftDocs/msteams-docs/pull/10598

### Further outdated Resources
The following examples are also outdated since they assume the initiating client is always teams client (msteams://) but could be M365 and Outlook mobile / new Outlook Windows Desktop app
* https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/a73ee669a4592e5f82f385b69c297648285e2caa/samples/tab-external-auth/csharp/TabExternalAuth/Views/Auth/GoogleEnd.cshtml#L22
* https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/a73ee669a4592e5f82f385b69c297648285e2caa/samples/tab-external-auth/nodejs/src/components/ClosePopup.js#L15

### CODE REVIEW

**Comment by onmomo** [comment]
> @onmomo please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="Starmind"]

**Comment by onmomo** [comment]
@v-vanv @mobajemu-msft please have a look.

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md](https://github.com/MicrosoftDocs/msteams-docs/blob/6aff6d543cc08c28ac5e451753e918d366e89e97/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-oauth-provider.md) | 92 | 15 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/9cee0ba4-52aa-42b7-bda2-bafe593ba1e5) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by mobajemu-msft** [comment]
> @v-vanv @mobajemu-msft please have a look.

Hey @Wajeed-msft would you mind taking a look as well? Thank you.

File: msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md [hunk 1]
```diff
@@ -59,30 +59,31 @@ The following image provides the flow to add authentication to external browsers
...
+   |`authId` |The request-id Teams creates for this specific authentication request that needs to be sent back to the client through a deep link.|
+   |`hostRedirectUrl` |The deep link containing the url scheme of the initiating client to redirect back to after the completion of authentication|
 
     > [!TIP]
-    > The app can marshal `authId`, `oauthRedirectMethod` in the OAuth `state` query parameter while generating the login URL for the OAuthProvider. The `state` contains the passed `authId` and `oauthRedirectMethod`, when OAuthProvider redirects back to the server and the app uses the values for sending authentication response back to Teams as described in step 6.
+    > The app can marshal `authId`, `oauthRedirectMethod` and `hostRedirectUrl` in the OAuth `state` query parameter while generating the login URL for the OAuthProvider. The `state` contains the passed `authId`, `oauthRedirectMethod` and `hostRedirectUrl`, when OAuthProvider redirects back to the server and the app uses the values for sending authentication response back to the initiating client as described in step 6.
```
Comment (v-sdhakshina) [inline msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md:79]:
```suggestion
    > The app can marshal `authId`, `oauthRedirectMethod`, and `hostRedirectUrl` in the OAuth `state` query parameter while generating the login URL for the OAuthProvider. The `state` contains the passed `authId`, `oauthRedirectMethod`, and `hostRedirectUrl`, when OAuthProvider redirects to the server and the app uses the values for sending authentication response back to the initiating client as described in step 6.
```

**Review by v-sdhakshina — CHANGES_REQUESTED** [review]

File: msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md [hunk 2]
```diff
@@ -59,30 +59,31 @@ The following image provides the flow to add authentication to external browsers
...
+1. The Teams clients open the URL in an external browser after automatically replacing the placeholders for `oauthRedirectMethod`, `authId` and `hostRedirectUrl` with suitable values.
 
    **Example**
 
    ```http
-    https://3p.app.server/auth?oauthRedirectMethod=deeplink&authId=1234567890 
+    https://3p.app.server/auth?oauthRedirectMethod=deeplink&authId=1234567890&hostRedirectUrl=msteams://teams.microsoft.com/l/auth-callback?authId=1234567890&result={result} 
    ```
 
 1. The third-party app server responds. The third-party app server receives and saves the `url` with the following two query parameters:
```
Comment (vikramtha) [inline msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md:70]: #
```suggestion
1. The third-party app server responds. The third-party app server receives and saves the `url` with the following three query parameters:
```

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md](https://github.com/MicrosoftDocs/msteams-docs/blob/b4a7a39adae31c9380369e205093f5b6651a6b4f/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-oauth-provider.md) | 92 | 15 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/624315e5-40cd-4268-ad10-4e553210aa0a) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by v-sdhakshina** [comment]
@onmomo I've reviewed the PR and added suggestions, request you to update the doc.

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md](https://github.com/MicrosoftDocs/msteams-docs/blob/1483530008c8805260ba11c7165b8ce3858198a3/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-oauth-provider.md) | 92 | 15 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/924ae51a-86f6-46e5-8390-02a4e6936b93) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md](https://github.com/MicrosoftDocs/msteams-docs/blob/a8778f23dd8d6ae4911b48b836ce8a729aa4ea06/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-oauth-provider.md) | 94 | 14 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/f87272e4-8383-46e4-8adc-349082892426) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Review by v-sdhakshina — APPROVED** [review]

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine.

**Review by v-deabira — APPROVED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md](https://github.com/MicrosoftDocs/msteams-docs/blob/6860a22fdec2d596a1facbff03bc1af255a3bdbe/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-oauth-provider.md) | 94 | 14 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/368709ef-23d9-471b-b55a-2cc644287d70) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Review by surbhigupta12 — APPROVED** [review]

**Review by surbhigupta12 — APPROVED** [review]

**Pull request merged**